### PR TITLE
feature: supported `priority` for route.

### PR DESCRIPTION
**Pull request by membphis**
- [x] doc

### CODE REVIEW

File: doc/admin-api-cn.md [hunk 1]
```diff
@@ -57,6 +57,7 @@
 |remote_addr|可选 |匹配规则|客户端请求 IP 地址: `192.168.1.101`、`192.168.1.102` 以及 CIDR 格式的支持 `192.168.1.0/24`。特别的，APISIX 也完整支持 IPv6 地址匹配：`::1`，`fe80::1`, `fe80::1/64` 等。|"192.168.1.0/24"|
 |remote_addrs|可选 |匹配规则|列表形态的 `remote_addr`，表示允许有多个不同 IP 地址，符合其中任意一个即可。|{"127.0.0.1", "192.0.0.0/8", "::1"}|
 |methods  |可选 |匹配规则|如果为空或没有该选项，代表没有任何 `method` 限制，也可以是一个或多个的组合：`GET`, `POST`, `PUT`, `DELETE`, `PATCH`, `HEAD`, `OPTIONS`，`CONNECT`，`TRACE`。|{"GET", "POST"}|
+|priority  |可选 |匹配规则|如果不同路由包含相同 `uri` ，根据属性 `priority` 确定哪个 `route` 被优先匹配。默认值为 `0`.|priority = 10|
```
Comment (juzhiyuan) [inline doc/admin-api-cn.md]: image

Please remove this space

File: doc/admin-api.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ Table of contents
 |remote_addr|False |Match Rules|The client requests an IP address: `192.168.1.101`, `192.168.1.102`, and CIDR format support `192.168.1.0/24`. In particular, APISIX also fully supports IPv6 address matching: `::1`, `fe80::1`, `fe80::1/64`, etc.|"192.168.1.0/24"|
 |remote_addrs|False |Match Rules|The `remote_addr` in the form of a list indicates that multiple different IP addresses are allowed, and match any one of them.|{"127.0.0.1", "192.0.0.0/8", "::1"}|
 |methods  |False |Match Rules|If empty or without this option, there are no `method` restrictions, and it can be a combination of one or more: `GET`,`POST`,`PUT`,`DELETE`,`PATCH`, `HEAD`,`OPTIONS`,`CONNECT`,`TRACE`.|{"GET", "POST"}|
+|priority  |False |Match Rules|If different routes contain the same `uri`, determine which route is matched first based on the attribute` priority`. Default value is `0`.|priority = 10|
```
Comment (juzhiyuan) [inline doc/admin-api.md]: image

Comment (juzhiyuan) [reply doc/admin-api.md]: The default value is xxx

File: doc/admin-api-cn.md [hunk 3]
```diff
@@ -57,6 +57,7 @@
 |remote_addr|可选 |匹配规则|客户端请求 IP 地址: `192.168.1.101`、`192.168.1.102` 以及 CIDR 格式的支持 `192.168.1.0/24`。特别的，APISIX 也完整支持 IPv6 地址匹配：`::1`，`fe80::1`, `fe80::1/64` 等。|"192.168.1.0/24"|
 |remote_addrs|可选 |匹配规则|列表形态的 `remote_addr`，表示允许有多个不同 IP 地址，符合其中任意一个即可。|{"127.0.0.1", "192.0.0.0/8", "::1"}|
 |methods  |可选 |匹配规则|如果为空或没有该选项，代表没有任何 `method` 限制，也可以是一个或多个的组合：`GET`, `POST`, `PUT`, `DELETE`, `PATCH`, `HEAD`, `OPTIONS`，`CONNECT`，`TRACE`。|{"GET", "POST"}|
+|priority  |可选 |匹配规则|如果不同路由包含相同 `uri`，根据属性 `priority` 确定哪个 `route` 被优先匹配，默认值为 0。|priority = 10|
```
Comment (moonming) [inline doc/admin-api-cn.md]: Is there a range? Is the larger the value the higher the priority?

Comment (moonming) [reply doc/admin-api-cn.md]: What would happen if the two priorities were the same?

File: lua/apisix/schema_def.lua [hunk 4]
```diff
@@ -283,6 +283,7 @@ _M.route = {
             uniqueItems = true,
         },
         desc = {type = "string", maxLength = 256},
+        priority = {type = "integer", minimum = 1},
```
Comment (moonming) [inline lua/apisix/schema_def.lua]: the minimum value is 1, but the default value is 0?

File: doc/admin-api-cn.md (hunk 3 above)
Comment (membphis) [reply doc/admin-api-cn.md]: Unable to determine which route was matched.

File: lua/apisix/schema_def.lua (hunk 4 above)
Comment (membphis) [reply lua/apisix/schema_def.lua]: Keep only the default value to 0.

**Pull request merged**